src/server/main_service.cc

```diff
@@ -1087,17 +1087,16 @@ void Service::Shutdown() {
 }
 
 OpResult<KeyIndex> Service::FindKeys(const CommandId* cid, CmdArgList args) {
-  if (!cid->IsShardedPSub()) {
-    return DetermineKeys(cid, args);
-  }
```

Comment:
If that's a top level if, then something's very odd. I understand that it breaks the rules, but we always properly work with abstractions to make the code simpler, see no-key-transactional commands, etc 🙄